File: PadrõesEstruturados/Adapter/Adapter01.js
// Interface do Cliente
class Taregt{
    
    request(){
    }
}

// Cliente
class Cliente{
    constructor(taregt){
        this.taregt = taregt;
    }

    criarRequest(){
        console.log("Cliente Fazendo Uma Requisição...")
    }
}

// Serviço Existente
class Adaptee{
    requestEspecifico(){
        console.log("Requisição Específica do Adaptee...")
    }
}

// Adaptador
class Adapter{
    constructor(adaptee){
        this.adaptee = adaptee;
    }

    request(){
        this.adaptee.requestEspecifico();
    }
}

// Uso do Adaptador:
const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const cliente = new Cliente(adapter);

cliente.criarRequest();

File: PadrõesEstruturados/Adapter/Adapter01.py
# Interface do Cliente
class Target:
    def request(self):
        pass

# Cliente
class Cliente:
    def __init__(self, target):
        self.target = target

    def criarRequest(self):
        print("Cliente fazendo uma requisição")
        self.target.request()

# Serviço Existente:
class Adaptee:
    def requestEspecifico(self):
        print("Requisição Específica do Adaptee")

# Adaptador
class Adapter:
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.requestEspecifico()

# Utilização do Adaptador:
adaptee = Adaptee()
adapter = Adapter(adaptee)
cliente = Cliente(adapter)

cliente.criarRequest()